### PR TITLE
Create release artifacts for Stargate v2 APIs

### LINKED ISSUE #2052: Unify developer docs for Quarkus-based APIs
The [Docs API README](https://github.com/stargate/stargate/blob/v2.0.0/apis/sgv2-docsapi/README.md) contains useful information about running the application and integration tests.

This will (hopefully) be similar for the REST and GraphQL APIs once they are migrated to Quarkus, so we should pull that information up in `apis/README.md`.

### LINKED ISSUE #2380: Include Stargate v2 APIs in release artifacts
Executables for the Stargate v2 APIs are not included in the release artifacts. The `stargate-jars.zip` includes jar files for the coordinator and the `starctl` script, but not the APIs. 

This makes it difficult for those who wish to deploy Stargate v2 without using Docker containers.

### PR DESCRIPTION
**Pull request by jeffreyscarpenter**
Publish a separate zip file for each Stargate v2 API as a release asset, with each zip containing the uber-jar and a script to start it. The script checks for required JDK.

Fixes #2380 
Fixes #2052

**Checklist**
- [x] Changes manually tested
- [ ] Automated Tests added/updated
- [x] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Comment by jeffreyscarpenter** [comment]
Still testing portions of this but wanted to give @ivansenic a chance to take a look at the basic concept.

File: apis/sgv2-docsapi/start-docsapi.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+set -e
+
+# Startup script for Stargate Document API
+#   This script can be used to start the Document API with an uber-jar.
+#   Any Java properties/options provided as arguments to this script are passed along to the `java` command.
```
Comment (ivansenic) [inline apis/sgv2-docsapi/start-docsapi.sh:6]: passing props does not work:

```
[13:04:53] ise:sgv2-docsapi <jeff/api-release-artifacts>$ ./start-docsapi.sh -Dstargate.data-store.ignore-bridge=true
   _____ __                         __            ____                       ___    ____  ____
  / ___// /_____ __________ _____ _/ /____       / __ \____  __________     /   |  / __ \/  _/
  \__ \/ __/ __ `/ ___/ __ `/ __ `/ __/ _ \     / / / / __ \/ ___/ ___/    / /| | / /_/ // /
 ___/ / /_/ /_/ / /  / /_/ / /_/ / /_/  __/    / /_/ / /_/ / /__(__  )    / ___ |/ _____/ /
/____/\__/\__,_/_/   \__, /\__,_/\__/\___/    /_____/\____/\___/____/    /_/  |_/_/   /___/
                    /____/

                                                               Powered by Quarkus 2.15.3.Final
INFO  [main] 2023-01-19 13:05:37,738 OpenApiRecorder.java:48 - CORS filtering is disabled and cross-origin resource sharing is allowed without restriction, which is not recommended in production. Please configure the CORS filter through 'quarkus.http.cors.*' properties. For more information, see Quarkus HTTP CORS documentation
WARN  [main] 2023-01-19 13:05:37,764 GrpcServerRecorder.java:108 - Using legacy gRPC support, with separate new HTTP server instance. Switch to single HTTP server instance usage with quarkus.grpc.server.use-separate-server=false property
INFO  [vert.x-eventloop-thread-1] 2023-01-19 13:05:37,897 GrpcServerRecorder.java:223 - gRPC Server started on 0.0.0.0:9000 [SSL enabled: false]
INFO  [main] 2023-01-19 13:05:37,930 Channels.java:216 - Creating Netty gRPC channel ...
INFO  [main] 2023-01-19 13:05:37,962 DataStorePropertiesConfiguration.java:59 - DataStoreConfig.ignoreBridge() == false, will try to fetch data store metadata using the Bridge
INFO  [main] 2023-01-19 13:05:38,083 DataStorePropertiesConfiguration.java:108 - Data store metadata fetch using Bridge failed (call #1/5), problem: (io.grpc.StatusRuntimeException) UNAVAILABLE: io exception
INFO  [main] 2023-01-19 13:05:40,916 DataStorePropertiesConfiguration.java:108 - Data store metadata fetch using Bridge failed (call #2/5), problem: (io.grpc.StatusRuntimeException) UNAVAILABLE: io exception
INFO  [main] 2023-01-19 13:05:44,000 DataStorePropertiesConfiguration.java:108 - Data store metadata fetch using Bridge failed (call #3/5), problem: (io.grpc.StatusRuntimeException) 
```

**Review by ivansenic — DISMISSED** [review]
I confirmed that Java version check works fine on my machine.. Listed some issues..

File: .github/workflows/release-v2.yml [hunk 2]
```diff
@@ -121,6 +121,100 @@ jobs:
...
+        uses: actions/upload-release-asset@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          upload_url: ${{needs.create-release.outputs.release-upload-url}}
+          asset_path: stargate-graphqlapi.zip
+          asset_name: stargate-graphqlapi.zip
+          asset_content_type: application/zip
+
   # publish coordinator JARs to the OSSRH
```
[the remark this answers is not in the document]
Comment (jeffreyscarpenter) [reply .github/workflows/release-v2.yml:218]: sure

File: apis/sgv2-docsapi/start-docsapi.sh (hunk 1 above)
Comment (jeffreyscarpenter) [reply apis/sgv2-docsapi/start-docsapi.sh:6]: fixing order. java props have to come before -jar option

File: apis/sgv2-docsapi/start-docsapi.sh [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  echo "Unable to locate sgv2-docsapi runner.jar in $DOCSAPI_DIR and subdirectories"
+  exit 3
+fi
+
+java -jar $DOCSAPI_JAR "$@"
+
+
+
+
+
```
[the remark this answers is not in the document]
Comment (jeffreyscarpenter) [reply apis/sgv2-docsapi/start-docsapi.sh:42]: will remove

File: .github/workflows/release-v2.yml [hunk 4]
```diff
@@ -121,6 +121,100 @@ jobs:
...
+        continue-on-error: true
+        run: |
+          JAVA_HOME=$JAVA_8 ./mvnw -B -ntp versions:set -DremoveSnapshot versions:commit
+          JAVA_HOME=$JAVA_8 ./mvnw -B -ntp clean install -DskipTests -pl bridge-proto
+
+      - name: Build APIs with Maven
+        run: |
+          cd apis/
+          JAVA_HOME=$JAVA_17 ./mvnw -B -ntp versions:set -DremoveSnapshot versions:commit
+          JAVA_HOME=$JAVA_17 ./mvnw -B -ntp clean package -Dquarkus.package.type=uber-jar -DskipTests
```
[the remark this answers is not in the document]
Comment (jeffreyscarpenter) [reply .github/workflows/release-v2.yml:180]: adding as requested

File: .github/workflows/release-v2.yml [hunk 5]
```diff
@@ -121,6 +121,100 @@ jobs:
...
+                <username>${ARTIFACTORY_USERNAME}</username>
+                <password>${ARTIFACTORY_PASSWORD}</password>
+             </server>
+           </servers>
+          </settings>
+          EOF
+
+      - name: Build bridge protos library
+        if: ${{ !inputs.skipPublish }}
+        continue-on-error: true
```
[the remark this answers is not in the document]
Comment (jeffreyscarpenter) [reply .github/workflows/release-v2.yml:171]: will remove

**Comment by jeffreyscarpenter** [comment]
Added commits to update READMEs with information about deployment options, in the process made some additional changes which will address #2052.

**Pull request merged**